### PR TITLE
Add coarsen.construct

### LINKED ISSUE #5454: Add coarsen.construct
Like `Rolling.construct` this would create a new reshaped array with extra dimensions that would usually be reduced over. It would be a light wrapper over `Variable._coarsen_reshape`

It make https://github.com/pydata/xarray/discussions/5119, and #2419 really easy without having to go through a MultiIndex+unstack operation which is somewhat arcane.
``` python
ds.coarsen(time=12).construct(dim="year").rename({"time": "month"})
```

### PR DESCRIPTION
**Pull request by dcherian**
- [x] Closes #5454
- [x] Tests added
- [x] Passes `pre-commit run --all-files`
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

Builds on #5474 

Here's an example
``` python
import numpy as np
import xarray as xr

ds = xr.Dataset(
    {
        "vart": ("time", np.arange(48)),
        "varx": ("x", np.arange(10)),
        "vartx": (("x", "time"), np.arange(480).reshape(10, 48)),
        "vary": ("y", np.arange(12)),
    },
    coords={"time": np.arange(48), "y": np.arange(12)},
)
ds.coarsen(time=12, x=5, boundary="trim").construct(
    {"time": ("year", "month"), "x": ("x", "x_reshaped")}
)
```

What do people think of this syntax:  `{"time": ("year", "month"), "x": ("x", "x_reshaped")`? Should we instead do `{"time": "month", "x": "x_reshaped"}` and have the user later rename the `x` or `time` dimension if they want?

### CODE REVIEW

File: xarray/tests/test_coarsen.py [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+            "vary": ("y", np.arange(12)),
+        },
+        coords={"time": np.arange(48), "y": np.arange(12)},
+    )
+
+    if dask and has_dask:
+        ds = ds.chunk({"x": 4, "time": 10})
+    with raise_if_dask_computes():
+        actual = ds.coarsen(time=12, x=5).construct(
+            {"time": ("year", "month"), "x": ("x", "x_reshaped")}
```
Comment (mathause) [inline xarray/tests/test_coarsen.py:328]: I don't understand why you pass a tuple here? Is that required?

**Review by mathause — COMMENTED** [review]
Looks good I left some minor comments. I have not looked at the moved tests.

File: xarray/tests/test_coarsen.py (hunk 1 above)
Comment (dcherian) [reply xarray/tests/test_coarsen.py:328]: Good q. We are taking one dimension and breaking it up in to two. Should we ask the user to pass in one name or two names? Here I've implemented the "two name" approach.

Comment (mathause) [reply xarray/tests/test_coarsen.py:328]: Ah yes it's reshaping the original dataset.

**Comment by dcherian** [comment]
This should be ready for review & merge

**Review by mathause — APPROVED** [review]
Looks good.

**Comment by dcherian** [comment]
Thanks for the review @mathause

**Pull request merged**